seafile-client 6.1.8 fails to build

## Issue description

seafile-client 6.1.8 fails to build at cmake configuration step with the following error message:

```
CMake Error at CMakeLists.txt:802 (QT5_USE_MODULES):
  Unknown CMake command "QT5_USE_MODULES".
```

### Steps to reproduce

run `nix-env -iA nixos.seafile-client`

## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.51, NixOS, 18.09pre144572.9720833743f (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0.4`
 - channels(root): `"nixos-18.09pre143771.a8c71037e04, nixpkgs-18.09pre144572.9720833743f, unstable-small-18.09pre144572.9720833743f"`
 - channels(nek0): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/applications/networking/seafile-client/default.nix
+++ b/pkgs/applications/networking/seafile-client/default.nix
@@ -5,14 +5,14 @@
 with stdenv.lib;
 
 stdenv.mkDerivation rec {
-  version = "6.1.8";
+  version = "6.2.1";
   name = "seafile-client-${version}";
 
   src = fetchFromGitHub {
     owner = "haiwen";
     repo = "seafile-client";
     rev = "v${version}";
-    sha256 = "0gy7jfxr5f8qvbqj80g7fzaw9b3vax750c4z5cr7f43rv99284pc";
+    sha256 = "02nzihs720pmgr5139s7fz88f7gm19dn1x2kqdpi81k7gdwqq0bp";
   };
 
   nativeBuildInputs = [ pkgconfig cmake makeWrapper ];

--- a/pkgs/development/libraries/libsearpc/default.nix
+++ b/pkgs/development/libraries/libsearpc/default.nix
@@ -1,15 +1,15 @@
-{stdenv, fetchurl, automake, autoconf, pkgconfig, libtool, python2Packages, glib, jansson}:
+{stdenv, fetchFromGitHub, automake, autoconf, pkgconfig, libtool, python2Packages, glib, jansson}:
 
-stdenv.mkDerivation rec
-{
-  version = "3.0.7";
-  seafileVersion = "6.1.0";
+stdenv.mkDerivation rec {
+  version = "3.0.8";
   name = "libsearpc-${version}";
 
-  src = fetchurl
-  {
-    url = "https://github.com/haiwen/libsearpc/archive/v${version}.tar.gz";
-    sha256 = "0fdrgksdwd4qxp7qvh75y39dy52h2f5wfjbqr00h3rwkbx4npvpg";
+  src = fetchFromGitHub {
+    owner = "haiwen";
+    repo = "libsearpc";
+    # Tag is missing: https://github.com/haiwen/libsearpc/commit/12a01268825e9c7e17794c58c367e3b4db912ad9
+    rev = "12a01268825e9c7e17794c58c367e3b4db912ad9";
+    sha256 = "00ck1hl1x0pn22q3ba32dq3ckc4nrsg58irsmrnmalqbsffhcim0";
   };
 
   patches = [ ./libsearpc.pc.patch ];
@@ -18,12 +18,13 @@ stdenv.mkDerivation rec
   buildInputs = [ automake autoconf libtool python2Packages.python python2Packages.simplejson ];
   propagatedBuildInputs = [ glib jansson ];
 
+  postPatch = "patchShebangs autogen.sh";
+
   preConfigure = "./autogen.sh";
 
   buildPhase = "make -j1";
 
-  meta =
-  {
+  meta = {
     homepage = https://github.com/haiwen/libsearpc;
     description = "A simple and easy-to-use C language RPC framework (including both server side & client side) based on GObject System";
     license = stdenv.lib.licenses.lgpl3;

--- a/pkgs/misc/seafile-shared/default.nix
+++ b/pkgs/misc/seafile-shared/default.nix
@@ -1,12 +1,14 @@
-{stdenv, fetchurl, which, autoreconfHook, pkgconfig, curl, vala, python, intltool, fuse, ccnet}:
+{stdenv, fetchFromGitHub, which, autoreconfHook, pkgconfig, curl, vala, python, intltool, fuse, ccnet}:
 
 stdenv.mkDerivation rec {
-  version = "6.1.0";
+  version = "6.2.1";
   name = "seafile-shared-${version}";
 
-  src = fetchurl {
-    url = "https://github.com/haiwen/seafile/archive/v${version}.tar.gz";
-    sha256 = "03zvxk25311xgn383k54qvvpr8xbnl1vxd99fg4ca9yg5rmir1q6";
+  src = fetchFromGitHub {
+    owner = "haiwen";
+    repo = "seafile";
+    rev = "v${version}";
+    sha256 = "199ivh6fbz4sgbz15mr6m8985yky5n3z6vpcpnh2ldjx4277gq68";
   };
 
   nativeBuildInputs = [ pkgconfig which autoreconfHook vala intltool ];
@@ -17,13 +19,6 @@ stdenv.mkDerivation rec {
     "--disable-server"
     "--disable-console"
   ];
-
-  postInstall = ''
-    # Remove seafile binary
-    rm -rf "$out/bin/seafile"
-    # Remove cli client binary
-    rm -rf "$out/bin/seaf-cli"
-  '';
 
   meta = with stdenv.lib; {
     homepage = https://github.com/haiwen/seafile;

